src/event_form/steps/event-form-step3.controller.js

```diff
@@ -226,87 +225,68 @@ function EventFormStep3Controller(
     $scope.selectedLocation = false;
     $scope.locationAutocompleteTextField = '';
     $scope.locationsSearched = false;
+    $scope.selectedCityObj = city;
 
     controller.stepUncompleted();
-    controller.getLocations(city);
   }
 
-  /**
-   * Get locations for Event.
-   * @returns {undefined}
-   * @param {Object} city
-   */
-  controller.getLocations = function (city) {
+  controller.getPlaces = function(filterValue) {
 
-    function showErrorAndReturnEmptyList () {
-      $scope.locationAutoCompleteError = true;
-      return [];
+    // exit code when the filterValue has < 3 characters
+    if (filterValue.length < 3) {
+      $scope.locationsSearched = false;
+      return;
```

Comment:
Good idea using an early return instead of wrapping the rest of the code below in one giant `if` 👍